xbmc/ServiceManager.cpp

```diff
@@ -250,6 +260,14 @@ void CServiceManager::DeinitStageOne()
   CScriptInvocationManager::GetInstance().UnregisterLanguageInvocationHandler(m_XBPython.get());
   m_XBPython.reset();
 #endif
+
+  m_advancedSettings->Clear();
+  m_advancedSettings.reset();
+}
+
+CAdvancedSettings &CServiceManager::GetAdvancedSettings()
+{
+  return *m_advancedSettings.get();
```

Comment:
This can be just `return *m_advancedSettings;`

Reply:
yep